main/SQLHelper.cpp

```diff
@@ -5632,6 +5665,22 @@ bool CSQLHelper::GetLastValue(const int HardwareID, const char* DeviceID, const
 	return result;
 }
 
+std::string  CSQLHelper::GetDeviceValue(const char* FieldName, const std::string& Idx)
+{
+	bool result = false;
+	std::vector<std::vector<std::string> > sqlresult;
+
+    sqlresult = safe_query("SELECT %s FROM DeviceStatus WHERE ( ID= %s )" , FieldName, Idx.c_str() );
+
+	if (!sqlresult.empty())
+	{
+		return sqlresult[0][0] ;
+	}
+
+	return "" ;
```

Comment:
Much better (cleaner) were `return std::string();`

Reply:
return type of this function is 'std::string' also I sugested to use that type. Has nothing to do with auto..

Reply:
ok.done !